test/compat.jl

```diff
@@ -0,0 +1,12 @@
+module TestCompat
```

Comment:
I wouldn't say this deserves a separate file. Can you just test this at the same time as `eachcol` and `eachrow`?